### PR TITLE
Feature/Add additional information to frame messages objects

### PR DESCRIPTION
**Pull request by acomodi**
This PR allows to collect more detailed information on the frame message object. In particular:

- Type --> whether the message is periodic, sporadic or mixed
- Delay --> if and how much delay can be applied to the message
- Start Delay --> time from startup to delay before sending the first message frame

### CODE REVIEW

File: src/cantools/database/can/message.py [hunk 1]
```diff
@@ -445,6 +451,14 @@ def send_type(self) -> Optional[str]:
 
         return self._send_type
 
+    @property
+    def type(self) -> Optional[str]:
+        """The message type, or ``None`` if unavailable.
+
+        """
```
Comment (andredami) [inline src/cantools/database/can/message.py]: Clarify docstring

File: src/cantools/database/can/message.py [hunk 2]
```diff
@@ -453,6 +467,22 @@ def cycle_time(self) -> Optional[int]:
 
         return self._cycle_time
 
+    @property
+    def delay_time(self) -> Optional[int]:
+        """The message delay time, or ``None`` if unavailable.
+
+        """
```
Comment (andredami) [inline src/cantools/database/can/message.py]: Clarify docstring

File: src/cantools/database/can/message.py [hunk 3]
```diff
@@ -453,6 +467,22 @@ def cycle_time(self) -> Optional[int]:
...
+
+        return self._delay_time
+
+    @property
+    def start_delay_time(self) -> Optional[int]:
+        """The start delay time, or ``None`` if unavailable.
+
+        """
```
Comment (andredami) [inline src/cantools/database/can/message.py]: Clarify docstring

File: src/cantools/database/can/formats/dbc.py [hunk 4]
```diff
@@ -1575,48 +1575,31 @@ def get_comment(frame_id_dbc):
...
         except KeyError:
             return None
-
-    def get_send_type(frame_id_dbc):
-        """Get send type for a given message.
-
-        """
+        
+    def get_attribute_value(frame_id_dbc, attribute):
+        """Get the attribute value of a given frame id"""
```
Comment (andredami) [inline src/cantools/database/can/formats/dbc.py]: Please, list all the attributes that are supported and their semantics.

File: src/cantools/database/can/formats/dbc.py [hunk 5]
```diff
@@ -1716,8 +1699,11 @@ def get_signal_groups(frame_id_dbc):
                     name=get_message_name(frame_id_dbc, message[2]),
                     length=int(message[4], 0),
                     senders=senders,
-                    send_type=get_send_type(frame_id_dbc),
-                    cycle_time=get_cycle_time(frame_id_dbc),
+                    send_type=get_attribute_value(frame_id_dbc, 'GenMsgSendType'),
+                    type=get_attribute_value(frame_id_dbc, 'Type'),
+                    cycle_time=get_attribute_value(frame_id_dbc, 'GenMsgCycleTime'),
+                    delay_time=get_attribute_value(frame_id_dbc, 'GenMsgDelayTime'),
+                    start_delay_time=get_attribute_value(frame_id_dbc, 'GenMsgStartDelayTime'),
```
Comment (andredami) [inline src/cantools/database/can/formats/dbc.py:1725]: Please, check that defaults are processed correctly for these attributes.

**Review by andredami — CHANGES_REQUESTED** [review]

File: src/cantools/database/can/formats/dbc.py
Comment (andredami) [inline src/cantools/database/can/formats/dbc.py:1]: Please, define the additional attributes as AttributeDefinition constants at module level

File: src/cantools/database/can/message.py (hunk 2 above)
Comment (andredami) [reply src/cantools/database/can/message.py]: Not quite sure, is this the case? (still, what would be the difference w.r.t. `start_delay_time`)
```python
        """The offset with respect to the message period. E.g.:
            - period -> 5 ms
            - delay -> 1 ms
            - scheduled sends -> n * period + delay (6 ms, 11 ms, 16 ms, ...)
        
        ``None`` if unavailable.
        """
```
or is this the case? (still, what would be the difference w.r.t. `cycle_time`?)
```python
        """The offset with respect to the message period. E.g.:
            - period -> 5 ms
            - delay -> 1 ms
            - scheduled sends -> n * (period + delay) (6 ms, 12 ms, 18 ms, ...)
        
        ``None`` if unavailable.
        """
```
I know it seems quite picky and not prioritary ATM, but I'm afraid this ambiguity could easily lead to code using these attributes not properly.

File: src/cantools/database/can/message.py (hunk 3 above)
Comment (andredami) [reply src/cantools/database/can/message.py]: See comment on `delay_time`

File: src/cantools/database/can/formats/dbc.py [hunk 6]
```diff
@@ -169,6 +169,22 @@
...
+    maximum=2**16-1)
+
+ATTRIBUTE_DEFINITION_GENMSGSTARTDELAYTIME = AttributeDefinition(
+    name='GenMsgStartDelayTime',
+    default_value=0,
+    kind='BO_',
+    type_name='INT',
+    minimum=0,
+    maximum=2**16-1)
+
```
Comment (andredami) [inline src/cantools/database/can/formats/dbc.py:187]: Missing `ATTRIBUTE_DEFINITION_GENMSGTYPE` (even if w/o default value, or other specifications).
If you don't have all the details but you think these details should be added after further investigation, let's add a TODO to the fields of the attribute (ENUM choiches I guess) that should be extended.

File: src/cantools/database/can/formats/dbc.py [hunk 7]
```diff
@@ -1575,48 +1591,37 @@ def get_comment(frame_id_dbc):
...
-
+        
+    def get_attribute_value(frame_id_dbc, attribute):
+        """Get the attribute value of a given frame id. Currently supported attributes:
+           - GenMsgSendType: specifies the whether a message is cyclic or sporadic
+           - Type: It is a further specification w.r.t. to GenMsgSendType that gives more details on whether the message is cyclic, sporadic, mixed, etc.
+           - GenMsgCycleTime: Defines the period of a cyclic message
+           - GenMsgDelayTime: Defines the offset w.r.t. to the cycle time
+           - GenMsgStartDelayTime: Defines the time interval between start-up and the first message
         """
```
Comment (andredami) [inline src/cantools/database/can/formats/dbc.py]: Seems like other attributes may be supported as well.
Maybe smth like:

```suggestion
        """Get the value of an attribute for a given frame id, possibly returning a default value if it exsists and the attribute is not available, without any ad-hoc processing.
        """
```

**Review by andredami — CHANGES_REQUESTED** [review]

File: src/cantools/database/can/formats/dbc.py (hunk 6 above)
Comment (acomodi) [reply src/cantools/database/can/formats/dbc.py:187]: added the TODO for now, we can check later on which is the correct default type for the `ATTRIBUTE_DEFINITION_TYPE` (as far as I see in the DBC files, the GENMSG prefix is absent)

**Review by andredami — APPROVED** [review]
LGTM

**Pull request merged**